cli/src/formatter.rs

```diff
@@ -522,6 +527,58 @@ impl<W: Write> Drop for ColorFormatter<W> {
     }
 }
 
+pub struct DebugFormatter<W: Write> {
+    formatter: ColorFormatter<W>,
+}
+
+impl<W: Write> DebugFormatter<W> {
+    pub fn new(output: W, rules: Arc<Rules>) -> Self {
+        Self {
+            formatter: ColorFormatter::new(output, rules),
+        }
+    }
+
+    pub fn for_config(output: W, config: &config::Config) -> Result<Self, config::ConfigError> {
+        let rules = rules_from_config(config)?;
+        Ok(Self::new(output, Arc::new(rules)))
```

Comment:
nit: forward to `ColorFormatter::for_config()`?

Alternatively, `DebugFormatter::new()` could take `ColorFormatter` as an argument, or `debug: bool` flag could be added to `ColorFormatter`.

Reply:
> Alternatively, `DebugFormatter::new()` could take `ColorFormatter` as an argument, or `debug: bool` flag could be added to `ColorFormatter`.

This is what I had initially, and I wasn't sure if that was wanted, because then the `Write` imply for `ColorFormatter` would have to be changed instead.

Not sure if this is more desirable.

Reply:
> > Alternatively, `DebugFormatter::new()` could take `ColorFormatter` as an argument, or `debug: bool` flag could be added to `ColorFormatter`.
> 
> This is what I had initially, and I wasn't sure if that was wanted, because then the `Write` imply for `ColorFormatter` would have to be changed instead.

Based on the code size, my feeling is that it's okay to be inlined in the `ColorFormatter`. `DebugFormatter` depends on the `ColorFormatter` internals anyway.

We can instead add `DebugFormatter` that tracks `labels` by itself (so can be combined with `PlainTextFormatter`), but I don't know if we'll need it.